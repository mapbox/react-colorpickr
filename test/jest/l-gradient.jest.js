import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as testCases from '../test-cases/l-gradient-test-cases';

describe('LGradient', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe(testCases.opacity.description, () => {
    beforeEach(() => {
      testCase = testCases.opacity;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
