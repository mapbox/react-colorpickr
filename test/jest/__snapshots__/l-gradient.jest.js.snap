// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LGradient basic renders 1`] = `
<div>
  <div
    key="1"
    style={Object {}}
  />
  <div
    key="2"
    style={
      Object {
        "background": "linear-gradient(to bottom, rgba(255,255,255,0) 0%, rgb(128,128,128) 100%)",
      }
    }
  />
  <div
    key="3"
    style={
      Object {
        "background": "rgb(255,255,255)",
        "opacity": -1,
      }
    }
  />
  <div
    key="4"
    style={
      Object {
        "background": "rgb(0,0,0)",
        "opacity": -1,
      }
    }
  />
</div>
`;

exports[`LGradient opacity renders 1`] = `<noscript />`;
